spring-data-jpa/src/main/java/org/springframework/data/jpa/repository/query/JSqlParserQueryEnhancer.java

```diff
@@ -449,10 +465,10 @@ public Set<String> getJoinAliases() {
 	 * @param query the query to parse
 	 * @return the parsed query
 	 */
-	private static Select parseSelectStatement(String query) {
+	private static <T extends Statement> T parseSelectStatement(String query, Class<T> classOfT) {
```

Comment:
Thats a clever solution! Make it more flexible in the future.
Personally I would also add a "default" version where we know it will be a `Select`. Because as you can see above we often call `parseSelectStatement(query.getQueryString(), Select.class);`.

```java
       /**
	 * Parses a query string with JSqlParser.
	 *
	 * @param query the query to parse
	 * @return the parsed query
	 */
	private Select parseSelectStatement(String query) {
		return parseSelectStatement(query,Select.class);
	}
```